File: Problems/0203-remove-linked-list-elements.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

#leetcode sol starts from here
class Solution(object):
    def removeElements(self, head, val):
        dummy = ListNode(0, head)
        curr = dummy

        while curr.next:
            if curr.next.val == val:
                curr.next = curr.next.next
            else:
                curr = curr.next

        return dummy.next

File: Problems/0203-remove-linked-list-elements.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here

struct ListNode {
    int val;
    ListNode* next;

    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode* next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode dummy(0, head);
        ListNode* curr = &dummy;

        while (curr->next) {
            if (curr->next->val == val)
                curr->next = curr->next->next;
            else
                curr = curr->next;
        }

        return dummy.next;
    }
};